bibtex-completion.el

```diff
@@ -478,6 +493,23 @@ for string replacement."
 (defvar bibtex-completion-cached-notes-keys nil
   "A cache storing notes keys obtained when the bibliography was last parsed.")
 
+(defun bibtex-completion--read ()
+  "Select BibTeX entries in completion system."
+  ;; define a completion function that defaults to completing-read, but can be overridden
+  (list (cdr (assoc (completing-read-bibtex--read) (bibtex-completion--get-candidates)))))
+
+(defun bibtex-completion--completing-read ()
+  "Read bibtex-completion entries for completion using completing-read."
+  (bibtex-completion-init)
+  (completing-read
+   "BibTeX entries: "
+   (bibtex-completion--get-candidates)))
```

Comment:
@minad - no rush, but how do I fix this to add back the completion category?

Here, for comparison was the original code from #355 that I started with, but I needed to simplify it a lot (I guess too much?) to get working here.

https://github.com/tmalsburg/helm-bibtex/pull/355/files#diff-43b58707c1adce5c6f3e1ed8d97ea5665182d94767472f7c196d630ff3ca0c3cR32-R37

Reply:
Attempted fix here.

https://github.com/tmalsburg/helm-bibtex/pull/361/commits/e5e4114691fec0dcca41d4577b7f16d44968049c

Reply:
Sorry, @bdarcus I didn't follow closely what is going on here. I don't understand what you are attempting here, the PR by @mtreca has not been merged yet? The prototype by @mtreca using the completion category did work before, didn't it? The commit e5e4114 seems to be correct.

Reply:
> I don't understand what you are attempting here, the PR by @mtreca has not been merged yet?

I'm providing this as an alternative PR, which @mtreca suggested, to see what modifying bibtex-completion itself would entail, and what impact it would have. 

I think it demonstrates what you were earlier arguing; that this is a better approach.

And it helps me understand better the code in doing so, regardless of which path we take.

> The prototype by @mtreca using the completion category did work before, didn't it?

Yes.

Reply:
I guess, @minad, the best way to debug this to come up with a minimal config file and use `emacs -Q`?

Any chance the selectrum et al dev community has come up with a standard minimal config for these purposes, perhaps that's posted on one of your wikis?

Reply:
@oantolin - if you can, could you please take a look at this?

It should work, but embark isn't picking up the keybindings.